Source/Scene/GroundPrimitive.js

```diff
@@ -0,0 +1,674 @@
+            this._colorPassCommands = new Array(commandsLength);
+            this._pickCommands = new Array(commandsLength);
+
+            for (i = 0; i < commandsLength; ++i) {
+                var primitiveCommand = primitiveCommandList[i];
+
+                this._stencilPreloadPassCommands[i] = new DrawCommand({
+                    primitiveType : primitiveCommand.primitiveType,
+                    vertexArray : primitiveCommand.vertexArray,
+                    renderState : this._rsStencilPreloadPass,
+                    shaderProgram : this._sp,
+                    uniformMap : primitiveCommand.uniformMap,
+                    boundingVolume : primitiveCommand.boundingVolume,
+                    owner : this,
+                    modelMatrix : Matrix4.IDENTITY,
+                    pass : Pass.GROUND
+                });
+
+                this._stencilDepthPassCommands[i] = new DrawCommand({
+                    primitiveType : primitiveCommand.primitiveType,
+                    vertexArray : primitiveCommand.vertexArray,
+                    renderState : this._rsStencilDepthPass,
+                    shaderProgram : this._sp,
+                    uniformMap : primitiveCommand.uniformMap,
+                    boundingVolume : primitiveCommand.boundingVolume,
+                    owner : this,
+                    modelMatrix : Matrix4.IDENTITY,
+                    pass : Pass.GROUND
+                });
+
+                this._colorPassCommands[i] = new DrawCommand({
+                    primitiveType : primitiveCommand.primitiveType,
+                    vertexArray : primitiveCommand.vertexArray,
+                    renderState : this._rsColorPass,
+                    shaderProgram : this._sp,
+                    uniformMap : primitiveCommand.uniformMap,
+                    boundingVolume : primitiveCommand.boundingVolume,
+                    owner : this,
+                    modelMatrix : Matrix4.IDENTITY,
+                    pass : Pass.GROUND
+                });
+
+                this._pickCommands[i] = new DrawCommand({
+                    primitiveType : primitiveCommand.primitiveType,
+                    vertexArray : primitiveCommand.vertexArray,
+                    renderState : this._rsPickPass,
+                    shaderProgram : this._spPick,
+                    uniformMap : primitiveCommand.uniformMap,
+                    boundingVolume : primitiveCommand.boundingVolume,
+                    owner : this,
+                    modelMatrix : Matrix4.IDENTITY,
+                    pass : Pass.GROUND
+                });
+            }
+        }
+
+        var stencilPreloadCommands = this._stencilPreloadPassCommands;
+        var stencilDepthPassCommands = this._stencilDepthPassCommands;
+        var colorPassCommands = this._colorPassCommands;
+        var pickCommands = this._pickCommands;
+
+        length = primitiveCommandList.length;
+        for (j = 0; j < length; ++j) {
```

Comment:
This approach of wrapping `_primitive` and then calling its `update` to pull out some data for our commands seems a bit shady.

Do you think we could reasonably change `Primitive` to add any hooks we need for overriding its command creation so then `GroundPrimitive` becomes a trivial wrapper over `Primitive` with just a few plugin points?

Would that be better or worse?